Source/msg.cpp

```diff
@@ -1862,24 +1862,25 @@ void run_delta_info()
 void DeltaExportData(int pnum)
 {
 	if (sgbDeltaChanged) {
-		int size;
-		std::unique_ptr<byte[]> dst { new byte[sizeof(DLevel) + 1] };
-		byte *dstEnd;
 		for (int i = 0; i < NUMLEVELS; i++) {
-			dstEnd = &dst[1];
+			std::unique_ptr<byte[]> dst { new byte[sizeof(DLevel) + 1] };
+			byte *dstEnd = &dst.get()[1];
 			dstEnd = DeltaExportItem(dstEnd, sgLevels[i].item);
 			dstEnd = DeltaExportObject(dstEnd, sgLevels[i].object);
 			dstEnd = DeltaExportMonster(dstEnd, sgLevels[i].monster);
-			size = CompressData(dst.get(), dstEnd);
-			dthread_send_delta(pnum, static_cast<_cmd_id>(i + CMD_DLEVEL_0), dst.get(), size);
+			int size = CompressData(dst.get(), dstEnd);
+			dthread_send_delta(pnum, static_cast<_cmd_id>(i + CMD_DLEVEL_0), std::move(dst), size);
 		}
-		dstEnd = &dst[1];
+
+		std::unique_ptr<byte[]> dst { new byte[MAXPORTAL * sizeof(DPortal) + MAXQUESTS * sizeof(MultiQuests) + 1] };

```

Comment:
Instead of `MAXQUESTS`, you can use `MAXMULTIQUESTS` since only the multiplayer quests get copied. In fact, I'd think you should be able to simply use `sizeof(DJunk) + 1` instead of breakign it out by portals and quests.